### PR TITLE
Add new parameters to findings API in security analytics

### PR DESCRIPTION
**Pull request by riysaxen-amzn**
### Description
Security Analytics Findings API Enhancements

### Issues Resolved
Closes https://github.com/opensearch-project/documentation-website/issues/6135

### Checklist
- [ ] By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license and subject to the [Developers Certificate of Origin](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

**Comment by hdhalter** [comment]
Thanks so much for submitting this PR, @riysaxen-amzn! Is it ready for review? I assume it should be backported to the lastest release (2.12)?

**Comment by riysaxen-amzn** [comment]
> Thanks so much for submitting this PR, @riysaxen-amzn! Is it ready for review? I assume it should be backported to the latest release (2.12)?

I couldn't test the changes in my local because of installation issues, other than it is ready for review. The changes shouldn't be backported to latest release 2.12

**Comment by hdhalter** [comment]
> > Thanks so much for submitting this PR, @riysaxen-amzn! Is it ready for review? I assume it should be backported to the latest release (2.12)?
> 
> I couldn't test the changes in my local because of installation issues, other than it is ready for review. The changes shouldn't be backported to latest release 2.12

OK, are these changes coming in 2.13?

**Comment by riysaxen-amzn** [comment]
> > > Thanks so much for submitting this PR, @riysaxen-amzn! Is it ready for review? I assume it should be backported to the latest release (2.12)?
> > 
> > 
> > I couldn't test the changes in my local because of installation issues, other than it is ready for review. The changes shouldn't be backported to latest release 2.12
> 
> OK, are these changes coming in 2.13?

yes they'll be in 2.13

File: _security-analytics/api-tools/alert-finding-api.md [hunk 1]
```diff
@@ -149,13 +149,230 @@ You can specify the following parameters when getting findings.
 
 Parameter | Description 
 :--- | :---
-`detector_id` | The ID of the detector used to fetch alerts. Optional when the `detectorType` is specified. Otherwise required.
-`detectorType` | The type of detector used to fetch alerts. Optional when the `detector_id` is specified. Otherwise required.
+`detector_id` | The ID of the detector used to fetch alerts. Optional
```
Comment (Naarcha-AWS) [inline _security-analytics/api-tools/alert-finding-api.md:152]:
```suggestion
`detector_id` | The ID of the detector used to fetch alerts. Optional.
```

**Review by Naarcha-AWS — CHANGES_REQUESTED** [review]

File: _security-analytics/api-tools/alert-finding-api.md [hunk 2]
```diff
@@ -149,13 +149,230 @@ You can specify the following parameters when getting findings.
...
+`detectorType` | The type of detector used to fetch alerts. Optional
 `sortOrder` | The order used to sort the list of findings. Possible values are `asc` or `desc`. Optional.
 `size` | An optional limit for the maximum number of results returned in the response. Optional.
+`startIndex` | The pagination indicator. Optional.
```
[the remark this answers is not in the document]
Comment (riysaxen-amzn) [reply _security-analytics/api-tools/alert-finding-api.md:156]: Similar to what we have for Alerts -> https://opensearch.org/docs/latest/security-analytics/api-tools/alert-finding-api/#get-alerts

File: _security-analytics/api-tools/alert-finding-api.md [hunk 3]
```diff
@@ -149,13 +149,230 @@ You can specify the following parameters when getting findings.
...
 Parameter | Description 
 :--- | :---
-`detector_id` | The ID of the detector used to fetch alerts. Optional when the `detectorType` is specified. Otherwise required.
-`detectorType` | The type of detector used to fetch alerts. Optional when the `detector_id` is specified. Otherwise required.
+`detector_id` | The ID of the detector used to fetch alerts. Optional.
+`detectorType` | The type of detector used to fetch alerts. Optional.
 `sortOrder` | The order used to sort the list of findings. Possible values are `asc` or `desc`. Optional.
 `size` | An optional limit for the maximum number of results returned in the response. Optional.
+`startIndex` | The pagination indicator. Optional.
+`detectionType` |  The detection rule type that dictates the retrieval type of the findings. When the detection type is `threat`, it fetches threat intelligence feeds. When it's `rule`, findings are fetched based on the detector's rule. Optional.
```
Comment (Naarcha-AWS) [inline _security-analytics/api-tools/alert-finding-api.md:157]:
```suggestion
`detectionType` |  The detection rule type that dictates the retrieval type for the findings. When the detection type is `threat`, it fetches threat intelligence feeds. When the detection type is `rule`, findings are fetched based on the detector's rule. Optional.
```

File: _security-analytics/api-tools/alert-finding-api.md [hunk 4]
```diff
@@ -149,13 +149,230 @@ You can specify the following parameters when getting findings.
...
+`severity` |  The severity of the detector rule used to fetch alerts. Severity can be critical|high|medium|low. Optional
```
Comment (Naarcha-AWS) [inline _security-analytics/api-tools/alert-finding-api.md:158]:
```suggestion
`severity` |  The severity of the detector rule used to fetch alerts. Severity can be `critical`, `high`, `medium`, or `low`. Optional
```

**Review by Naarcha-AWS — APPROVED** [review]

File: _security-analytics/api-tools/alert-finding-api.md [hunk 5]
```diff
@@ -149,13 +149,230 @@ You can specify the following parameters when getting findings.
...
 :--- | :---
-`detector_id` | The ID of the detector used to fetch alerts. Optional when the `detectorType` is specified. Otherwise required.
-`detectorType` | The type of detector used to fetch alerts. Optional when the `detector_id` is specified. Otherwise required.
+`detector_id` | The ID of the detector used to fetch alerts. Optional.
+`detectorType` | The type of detector used to fetch alerts. Optional.
 `sortOrder` | The order used to sort the list of findings. Possible values are `asc` or `desc`. Optional.
 `size` | An optional limit for the maximum number of results returned in the response. Optional.
+`startIndex` | The pagination indicator. Optional.
+`detectionType` |  The detection rule type that dictates the retrieval type for the findings. When the detection type is `threat`, it fetches threat intelligence feeds. When the detection type is `rule`, findings are fetched based on the detector's rule. Optional.
+`severity` |  The severity of the detector rule used to fetch alerts. Severity can be `critical`, `high`, `medium`, or `low`. Optional
```
Comment (natebower) [inline _security-analytics/api-tools/alert-finding-api.md:158]:
```suggestion
`severity` |  The severity of the detector rule used to fetch alerts. Severity can be `critical`, `high`, `medium`, or `low`. Optional.
```

**Review by natebower — COMMENTED** [review]
@riysaxen-amzn @Naarcha-AWS Just a period to be added. Thanks!

**Pull request merged**